projects/packages/account-protection/src/class-password-manager.php

```diff
@@ -95,26 +95,33 @@ public function validate_password_reset( \WP_Error $errors, $user ): void {
 	/**
 	 * Handle the profile update.
 	 *
-	 * @param int      $user_id The user ID.
-	 * @param \WP_User $old_user_data Object containing user data prior to update.
+	 * @param int                $user_id The user ID.
+	 * @param \WP_User|\stdClass $old_user_data Object containing user data prior to update.
 	 *
 	 * @return void
 	 */
-	public function on_profile_update( int $user_id, \WP_User $old_user_data ): void {
+	public function on_profile_update( int $user_id, $old_user_data ): void {
 		// phpcs:ignore WordPress.Security.NonceVerification
 		if ( isset( $_POST['action'] ) && $_POST['action'] === 'update' ) {
-			$this->save_recent_password_hash( $user_id, $old_user_data->user_pass );
+			// Only save if user_pass property exists and is not empty
+			if ( isset( $old_user_data->user_pass ) && '' !== $old_user_data->user_pass ) {
+				$this->save_recent_password_hash( $user_id, $old_user_data->user_pass );
+			}
 		}
 	}
 
 	/**
 	 * Handle the password reset.
 	 *
-	 * @param \WP_User $user The user.
+	 * @param \WP_User|\stdClass $user The user object.
 	 *
 	 * @return void
 	 */
-	public function on_password_reset( \WP_User $user ): void {
+	public function on_password_reset( $user ): void {
+		if ( ! isset( $user->ID ) || ! isset( $user->user_pass ) || empty( $user->user_pass ) ) {
```

Comment:
Could just do this if you want:
```suggestion
		if ( ! isset( $user->ID ) || empty( $user->user_pass ) ) {
```

Either way this won't save an old password of `0`. Not sure if we care.

Just to prevent inconsistencies down the road, the password check condition in `on_profile_update` vs. `on_password_reset` should probably be identical. Better yet, any reason to not just move this check directly into `save_recent_password_hash`?

Reply:
> Either way this won't save an old password of 0. Not sure if we care.

Since I'm not 100% familiar with the feature, I would prefer to keep it as much explicit as possible.
I updated the guard clause to also consider if it is an object:

https://github.com/Automattic/jetpack/pull/43045/files#diff-fe8256f355e8848c15a4c61a610b3b7c0ef832d25bdc0814547792476abfb04bR122

> Just to prevent inconsistencies down the road, the password check condition in on_profile_update vs. on_password_reset should probably be identical. Better yet, any reason to not just move this check directly into save_recent_password_hash?

This suggestion makes sense. I'd prefer to keep it separate for now, as I have some unmerged PRs related to this feature that I'd like to review before and avoid any potential code conflicts.

@tbradsha would you be okay by merging this and monitor it is all good and revisit this in a different PR?

Reply:
> I would prefer to keep it as much explicit as possible.

I'll note that it's identical logic with or without the `isset()`.

> I'd prefer to keep it separate for now

👍 

> would you be okay by merging this

Yes, I don't think we've broken things, and even if we missed an edge case it should prevent several failures. It's not clear if you understood my comment about a `0` password; though. Prior to this PR, I believe it would log the old password hash if the old password was falsy (i.e. `0`). Now it will not. I don't think it matters, but it is a change this PR introduces.

Reply:
> Yes, I don't think we've broken things, and even if we missed an edge case it should prevent several failures. It's not clear if you understood my comment about a 0 password; though. Prior to this PR, I believe it would log the old password hash if the old password was falsy (i.e. 0). Now it will not. I don't think it matters, but it is a change this PR introduces.

Thanks for the clarification! I don’t think there will be an issue with `0` either. I went ahead with `is_object()` and `empty()` to simplify things while keeping it safe 🙂. Let me know what you think! https://github.com/Automattic/jetpack/pull/43045/commits/2a437ce671302a6921569adfdde34a80480abfd5